client/state/account-recovery/actions.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * External dependencies
+ */
+import wpcom from 'lib/wp';
+
+/**
+ * Internal dependencies
+ */
+import {
+	ACCOUNT_RECOVERY_FETCH,
+	ACCOUNT_RECOVERY_FETCH_SUCCESS,
+	ACCOUNT_RECOVERY_FETCH_FAILED,
+} from 'state/action-types';
+
+export const accountRecoveryFetchSuccess = ( accountRecoverySettings ) => {
+	return {
+		type: ACCOUNT_RECOVERY_FETCH_SUCCESS,
+		accountRecoverySettings,
+	};
+};
+
+export const accountRecoveryFetchFailed = ( error ) => {
+	return {
+		type: ACCOUNT_RECOVERY_FETCH_FAILED,
+		error,
+	};
+};
+
+export const accountRecoveryFetch = () => ( dispatch ) => {
+	dispatch( { type: ACCOUNT_RECOVERY_FETCH } );
+
+	return new Promise( ( resolve, reject ) => {
```

Comment:
You shouldn't need to create your own `Promise` instance here, but you'd need to update the undocumented endpoint function to return its underlying `wpcom.req` call (which itself is a promise). I've seen this a few times now so put together #9116 to mass update all the undocumented endpoints.